### ISSUE TITLE
nghttpx client becomes slow after mac sleep

### ISSUE THREAD

**Issue by humiaozuzu** [nghttp2/nghttp2#714]
OS: macOS 10.12
I'm running nghttpx client on my mac as client and connecting to nghttpx server using H2 tunnel. And the speed is quite OK. But after I carry my macbook to home (system just sleep not shutdown), nghttpx client becomes quite slow. It takes about 30+s to open a webpage.

But after i stop the client and re-run in my terminal, it becomes quick as normal again.

Why it it like this and how can I help to debug the issue?

**Comment by tatsuhiro-t** [comment]
I'm not sure but this is probably an network related issue.  For some reason, nghttpx didn't know the connection was already broken, and keep re-sending packets.  We have similar issue when nghttpx is running in Android device, and we have a workaround to detect this kind of issue.  Perhaps, network configuration change makes connection to backend disrupted in some way.
Could you provide us nghttpx INFO level log when this happens?

**Comment by humiaozuzu** [comment]
@tatsuhiro-t I have just meet with slow network and I attached the logs before I restarted nghttpx
[nghttpx.log.zip](https://github.com/nghttp2/nghttp2/files/610822/nghttpx.log.zip)

**Comment by tatsuhiro-t** [comment]
When nghttpx writes the end of the log, it was slow, right?  Or is it recovered in some point of time?  If so, could you give us the date time when nghttpx was actually slow?
The log around the end of the file is not so strange to me.

**Comment by humiaozuzu** [comment]
Yes, from the end of the log. When I feels slow I stopped nghttpx and uploaded last lines of log

**Comment by tatsuhiro-t** [comment]
Thank you.  I think nghttpx might not write any log in that period.  Just waiting for something from peer, but because of network change, connection is lost, and answer will never come back.

**Comment by humiaozuzu** [comment]
So was this happen only on macOS and Android?

**Comment by tatsuhiro-t** [comment]
No, it can happen everywhere.  Devices using WiFi will see this happening more frequently since we can easily lost connection without TCP closing handshake when we go outside of WiFi range.
We plan to set timeout to detect this situation, but it is still a timeout so it takes some time to react to.

**Comment by humiaozuzu** [comment]
@tatsuhiro-t I've meet with another problem seems related to this.
My nghttpx client running after a while (10mins later), when downloading large files(eg. 100MB) the speed for first 10-20MBs is quite fast then the speed becomes slow(1-10KB/s) suddenly.

My nghttpx conf is
```
backend=test.com,443;;proto=h2;tls
frontend=127.0.0.1,1235;no-tls
workers=4
http2-proxy=yes
insecure=yes
add-x-forwarded-for=no
no-via=yes
no-ocsp=yes
npn-list=h2
tls-proto-list=TLSv1.2
backend-http2-window-size=4194303
backend-http2-connection-window-size=1073741823
```

My nghttpx is staticly compiled with following param

```
--enable-app --disable-silent-rules --disable-examples --disable-hpack-tools --disable-failmalloc --disable-python-bindings
--without-spdylay --without-libxml2
--with-boost=/usr/local/opt/boost --with-jemalloc
```

**Comment by tatsuhiro-t** [comment]
Does this occur only with nghttpx?   Do you have this issue without nghttpx?
